docs/en/03_Upgrading/01_Upgrading_project.md

```diff
@@ -338,36 +338,62 @@ You'll likely have some conflicts to resolve, whether you've updated your depend
 
 If the maintainer of the module is in the process of upgrading to SilverStripe 4, a development version of the module might be available. In some cases, it can be worthwhile to look up the repository of the module or to reach out to the maintainer.
 
 <div class="info" markdown="1">
-If you're going to install development version of third party modules, you should consider adding the following entries to your `composer.json` file.
+If you're going to install a development version of third party modules, you should consider adding the following entries to your `composer.json` file:
 
 ```json
 {
   // ...
   "minimum-stability": "dev",
-  "prefer-stable": true,
+  "prefer-stable": true
   // ...
 } 
 ```
 </div>
 
-To resolve a conflict you can either:
-* fork the affected module and upgrade it yourself. Don't forget to send a pull request to the original module!
-* Integrate the affected module into your project's codebase
-* Remove the module from your project, if it is not essential
+If no development release is available for SilverStripe 4, you can upgrade the module manually or remove the module from your project.
 
-To integrate a third party module in your project, remove it from your `composer.json` file and from your `.gitignore` file. Then track the module's codebase in your project source control. You'll need to upgrade the module's code to be compatible with SilverStripe 4. 
+#### Upgrading the module manually
+
+To upgrade an incompatible module yourself, follow these steps:
+
+1. Integrate the affected module into your project's codebase
+
+    1. Remove the module from your dependencies: `composer remove <package>` or through `composer.json`
+    2. Recompose again, this should now work without the conflicting module: `bin/upgrade-code recompose`
+    3. Clone the module repository into your project root: `git clone <repository>`
+
+<div class="info" markdown="1">
+Another approach is to remove the `_config.php` file from the module and remove the module folder from your project roots `.gitignore` so it becomes version tracked by your project.
+</div>
+
+2. Fork the affected module and upgrade it yourself
+
+    1. Fork the repository: `git fork --remote-name <remote>`
+    2. Upgrade the module so it works with version `4` of SilverStripe, commit and push your changes to your forked repository. See [Upgrading a module](./upgrading_module) for more information on how to upgrade a SilverStripe module.
 
 <div class="info" markdown="1">
-If you're taking the time to upgrade a third party module, consider doing a pull request against the original project so other developers can benefit from your work or releasing your fork as a seperate module.
+If your module is a part of a private VCS repository, this will break when upgrading.
+
+When running the upgrade command on the project it will not include root module folders, so you must run the upgrade command separately for each root module.
```

Comment:
```suggestion
When running the upgrade command on the project, it will not include root module folders. You must run the upgrade command separately for each root module.
```